contents/tutorials/react-charts.md

```diff
@@ -0,0 +1,454 @@
+---
+title: How to use React Charts to visualize analytics data (with examples)
+date: 2024-08-23
+author:
+ - ian-vanagas
+tags:
+ - product analytics
+---
+
+React Charts is a popular visualization and charting library for React. It provides a simplified set of performant charts to use with analytics data from PostHog.
+
+To provide examples of what you can do with it, we create a basic Next.js app, set it up to get data from PostHog's query API, and visualize it with React Charts.
+
+> **Don't want to do all the work of querying and visualizing?** You can always share or embed insights, dashboards, and more. See [sharing docs](/docs/product-analytics/sharing) for more.
+
+## Creating a Next.js app
+
+Because we need to query private data, we need a way to make server side requests. Next.js makes this easy (which is why we chose it).
+
+To create a Next.js app, run the following command, choose all the default options, and select the `app` router.
+
+```bash
+npx create-next-app@latest react-charts-example
+```
+
+This creates a `react-charts-example` folder with everything we need to get started.
+
+## Querying data from PostHog
+
+Assuming you have data in PostHog to query, the next step is to set up our query request to PostHog.
+
+> **Don't have data to query?** Check out [our guide to setting up analytics in Next.js](/tutorials/nextjs-app-directory-analytics) to start capturing some. Unlike other tutorials, we aren't going to install PostHog in this one.
+
+Start by creating a personal API key. You can do this by going to [personal API keys](https://us.posthog.com/settings/user-api-keys) in your project settings, clicking **Create personal API key**, giving it a label, choosing the **Performing analytics queries** scope (AKA query read), and clicking **Create key**.
+
+<ProductScreenshot
+  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/light_api_468581fdca.png" 
+  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/dark_api_a755bfb6ac.png"
+  alt="Creating a personal API key in PostHog" 
+  classes="rounded"
+/>
+
+Next, go into your `react-charts-example` folder, then the `app` folder, and create an `api` folder. In the `api` folder, create a `query.js` file. Here we set up a fetch request to PostHog's [query API endpoint](/docs/api/query) using your project ID (from your PostHog URL) and personal API key like this:
+
+```js
+// app/api/query.js
+export const fetchQuery = async (payload) => {
+  const projectId = "12345";
```

Comment:
Would be helpful to point ou where this is exactly. I know you mention in the top, but helpful to be a bit more prescriptive
```suggestion
  const projectId = "12345"; // e.g. if your PostHog dashboard url is https://us.posthog.com/project/12345/